testscommon/genesisMocks/runTypeComponentsStub.go

```diff
@@ -0,0 +1,118 @@
+package genesisMocks
+
+import (
+	"github.com/multiversx/mx-chain-go/process/block/preprocess"
+	"github.com/multiversx/mx-chain-go/process/coordinator"
+	"github.com/multiversx/mx-chain-go/process/smartContract/hooks"
+	"github.com/multiversx/mx-chain-go/process/smartContract/processProxy"
+	"github.com/multiversx/mx-chain-go/process/smartContract/processorV2"
+	"github.com/multiversx/mx-chain-go/process/smartContract/scrCommon"
+	"github.com/multiversx/mx-chain-go/sharding"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/state/factory"
+	"github.com/multiversx/mx-chain-go/testscommon/enableEpochsHandlerMock"
+	"github.com/multiversx/mx-chain-go/testscommon/hashingMocks"
+	"github.com/multiversx/mx-chain-go/testscommon/marshallerMock"
+)
+
+// RunTypeComponentsStub -
+type RunTypeComponentsStub struct {
+	BlockChainHookHandlerFactory  hooks.BlockChainHookHandlerCreator
+	TransactionCoordinatorFactory coordinator.TransactionCoordinatorCreator
+	SCResultsPreProcessorFactory  preprocess.SmartContractResultPreProcessorCreator
+	SCProcessorFactory            scrCommon.SCProcessorCreator
+	AccountCreator                state.AccountFactory
+	ShardCoordinatorFactory       sharding.ShardCoordinatorFactory
+	TxPreProcessorFactory         preprocess.TxPreProcessorCreator
+}
+
+// NewRunTypeComponentsStub -
+func NewRunTypeComponentsStub() *RunTypeComponentsStub {
+	blockChainHookHandlerFactory, _ := hooks.NewBlockChainHookFactory()
+	transactionCoordinatorFactory, _ := coordinator.NewShardTransactionCoordinatorFactory()
+	scResultsPreProcessorCreator, _ := preprocess.NewSmartContractResultPreProcessorFactory()
```

Comment:
All these factories should not return any error, but can be done in another task, please add one for it

Reply:
added task `MX-15415` and comment in `runTypeComponents.go`